core/services/pipeline/common.go

```diff
@@ -169,17 +171,38 @@ type JSONSerializable struct {
 	Null bool
 }
 
+// UnmarshalJSON implements custom unmarshaling logic
 func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 	if js == nil {
 		*js = JSONSerializable{}
 	}
-	return json.Unmarshal(bs, &js.Val)
+
+	if err := json.Unmarshal(bs, &js.Val); err != nil {
+		return err
+	}
+
+	// This "hack" is needed to solve the issue related to "\u0000" characters that should be stored in DB.
+	// PostgreSQL does not allow to store such characters so it should be HEX encoded.
+	if val, ok := js.Val.(string); ok {
+		if strings.HasPrefix(val, "hex:") {
+			val = strings.TrimPrefix(val, "hex:")
+			valRaw, err := hex.DecodeString(val)
+			if err != nil {
+				return err
+			}
+
+			js.Val = valRaw
+		}
+	}
```

Comment:
Some of the task params code uses `0x` to indicate a base64 encoded value: https://github.com/smartcontractkit/chainlink/blob/eb7b50980ad645809e3d796c6e63e01adb1e3dd0/core/services/pipeline/task_params.go#L221-L222

I think we could follow the same convention here.

Reply:
Good idea, but what is the original value is a HEX encoded string and no needed to decode it? 

I mean there could be two cases: 
1. Bytes array that should be HEX encoded and HEX decoded.
2. HEX encoded string so it should not be decoded in the unmarshal json function

Reply:
A hex encoded string is already a string: nothing to do

A []byte ought to be encoded by the default serialization of JSON

Reply:
That sounds good to me. Done.

Reply:
@samsondav I just realized it will not work. 

```
func (js JSONSerializable) MarshalJSON() ([]byte, error) {
	switch x := js.Val.(type) {
	case []byte:
		return json.Marshal(hex.EncodeToString(x))
	default:
		return json.Marshal(js.Val)
	}
}
```

`js.Val` could be a json string represented in bytes. No needed to hex encode JSON strings. It is kind of implicit logic.

Reply:
Solution could be the following 
```
func (js JSONSerializable) MarshalJSON() ([]byte, error) {
	switch x := js.Val.(type) {
	case []byte:
		if json.Valid(x) {
			return json.Marshal(string(x))
		} else {
			return json.Marshal(hex.EncodeToString(x))
		}
	default:
		return json.Marshal(js.Val)
	}
}
```

Reply:
I've been having similar problems with task inputs (https://github.com/smartcontractkit/chainlink/pull/4939), no type is stored alongside the raw data so it's hard to determine what to do when deserializing: is it a []byte that's base64 encoded or whether it should be decoded as a string

Reply:
@archseer but what if the original value is base64? we don't want to decode it in this case, I think.